File: 1913/solution.ts
function maxProductDifference(nums: number[]): number {
  let curr: number[] = [];

  for (let num of nums) {
    curr.push(num);
    curr.sort((a, b) => a - b);

    if (curr.length > 4) {
      curr.splice(2, 1);
    }
  }

  return curr[3] * curr[2] - curr[1] * curr[0];
}

File: 1913/solution.go
package main

import (
	"sort"
)

func maxProductDifference(nums []int) int {
    var curr []int
    for _, num := range nums {
        curr = append(curr, num)
        sort.Ints(curr)
        if len(curr) > 4 {
            curr = append(curr[:2], curr[3:]...) 
        }
    }

    return curr[3] * curr[2] - curr[1] * curr[0]
}